Highlight behavior regarding searchableAttributes (#1899)
meilisearch/documentation: reference/api/search.md

`attributesToHighlight` also highlights terms configured as [synonyms](/reference/api/settings.md#synonyms) and [stop words](/reference/api/settings.md#stop-words).
:::

::: warning

> `attributesToHighlight` will highlight all attributes added to the `attributesToHighlight` array even if they are not set as `searchableAttributes`.

Reviewer: ```suggestion
`attributesToHighlight` will highlight matches within all attributes added to the `attributesToHighlight` array, even if those attributes are not set as [`searchableAttributes`](/learn/configuration/displayed_searchable_attributes.md#searchable-fields).
```

Adds a `_matchesPosition` object to the search response that contains the location of each occurrence of queried terms across all fields. This is useful when you need more control than offered by our [built-in highlighting](#attributes-to-highlight). `showMatchesPosition` only works for strings, numbers, and arrays of strings and numbers.

::: warning

> `showMatchesPosition` will consider all attributes that contain words matching the query terms even if they are not set as `searchableAttributes`.

Reviewer 2: "consider" is a bit unclear here. Can we find a better verb, or group of words?

I'm contrasting it to the new warning on L#568, where "highlight" tells me what happens even if I haven't read anything else about that parameter.

Author: I can only think of "takes into account", but that's not helpful either 🤔

Reviewer 2: What about "returns match location"?

```suggestion
`showMatchesPosition` returns match location for all attributes that contain words matching the query terms even if they are not set as `searchableAttributes`.
```

> `showMatchesPosition` returns match location for all attributes that contain words matching the query terms even if they are not set as `searchableAttributes`.

Reviewer: ```suggestion
`showMatchesPosition` returns the location of matched query terms within all attributes, even attributes that are not set as `searchableAttributes`.
```
